metadata-ingestion/src/datahub/ingestion/source/unity/usage.py

```diff
@@ -0,0 +1,217 @@
+import json
+import logging
+import time
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Generic, Iterable, List, Optional, Set, TypeVar
+
+import pyspark
+from sqllineage.runner import LineageRunner
+
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.unity.config import UnityCatalogSourceConfig
+from datahub.ingestion.source.unity.proxy import UnityCatalogApiProxy
+from datahub.ingestion.source.unity.proxy_types import (
+    OPERATION_STATEMENT_TYPES,
+    Query,
+    StatementType,
+    TableReference,
+)
+from datahub.ingestion.source.unity.report import UnityCatalogReport
+from datahub.ingestion.source.usage.usage_common import UsageAggregator
+from datahub.metadata.schema_classes import OperationClass
+
+logger = logging.getLogger(__name__)
+
+
+TableMap = Dict[str, List[TableReference]]
+T = TypeVar("T", bound=object)
+
+
+@dataclass  # Dataclass over NamedTuple to support generic type annotations
+class GenericTableInfo(Generic[T]):
+    source_tables: List[T]
+    target_tables: List[T]
+
+
+StringTableInfo = GenericTableInfo[str]
+QueryTableInfo = GenericTableInfo[TableReference]
+
+
+@dataclass(eq=False)
+class UnityCatalogUsageExtractor:
+    config: UnityCatalogSourceConfig
+    report: UnityCatalogReport
+    proxy: UnityCatalogApiProxy
+    table_urn_builder: Callable[[TableReference], str]
+    user_urn_builder: Callable[[str], str]
+
+    def __post_init__(self):
+        self.usage_aggregator = UsageAggregator[TableReference](self.config)
+        self._spark_sql_parser: Optional[Any] = None
+
+    @property
+    def spark_sql_parser(self):
+        """Lazily initializes the Spark SQL parser."""
+        if self._spark_sql_parser is None:
+            spark_context = pyspark.SparkContext.getOrCreate()
```

Comment:
Do we install pyspark dependency for unity-catalog ? Its not explicitly mentioned in setup.py. Is it indirectly installed for databricks-cli ?

Reply:
Ah good catch, it's part of delta lake but not databricks